runtime/drivers/duckdb/duckdb.go

```diff
@@ -37,46 +45,111 @@ func (d driver) Open(dsn string) (drivers.Connection, error) {
 		"SET max_expression_depth TO 250",
 	}
 
+	metaConn, err := db.Connx(context.Background())
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Instead of having a `metaConn`, we should use `getConn` in the meta calls. Then it will use the `ctx`'s conn if there is one. 

Since the semaphore size is one less than the pool size, we know there will always be one connection extra in the pool for meta calls.